File: ExpenseSplitter.Backend/ExpenseSplitter.Api/Models/Expenses/UpdateExpenseModel.cs

using System;
using System.Collections.Generic;
using ExpenseSplitter.Api.Data;

namespace ExpenseSplitter.Api.Models.Expenses
{
    public class UpdateExpenseModel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public ExpenseType Type { get; set; }
        public DateTime PaidAt { get; set; }
        public int PayerId { get; set; }

        public List<ExpensePart> Parts { get; set; }
    }
}

File: ExpenseSplitter.Frontend/ExpenseSplitter.Web/src/app/models/expense/update-expense-model.ts
import { ExpenseTypeEnum } from 'src/app/data/expense-type';
import { ExpensePartModel } from './expense-part-model';

export class UpdateExpenseModel {
    id?: number;
    name: string;
    type: ExpenseTypeEnum;
    paidAt: Date;
    payerId: number;

    parts: ExpensePartModel[];
}